In [2]:
import pandas as pd
import numpy as np

Se encuentran datos aqui http://mineducacion.gov.co/sistemasdeinformacion/1735/w3-article-212400.html sobre:  
- Inscritos 
- admitidos
- matriculados

Cargo los datos del ICFES del año 2015 semestre 2 para comparar con las cantidades obtenidas en los nuevos archivos adquiridos de las Bases de Datos del ministerio de educacion

In [45]:
ICFES_2015 = pd.read_csv("../DatosFTPICFES/SABER11/SB11-BASES_DE_DATOS/SB11-20152-RGSTRO-CLFCCN-V3-0.txt", delimiter="|")

/usr/local/anaconda3/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2723: DtypeWarning: Columns (10,12,13,14,64) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [53]:
ICFES_2015.describe()

PERIODO      ESTU_EDAD  ESTU_PAIS_RESIDE    ESTU_ETNIA  \
count  573128.0  573100.000000     573128.000000  37477.000000   
mean    20152.0      17.863497          1.118944      9.358220   
std         0.0       4.357134          4.910093     18.542173   
min     20152.0       0.000000          1.000000      1.000000   
25%     20152.0      16.000000          1.000000      1.000000   
50%     20152.0      17.000000          1.000000      1.000000   
75%     20152.0      18.000000          1.000000     13.000000   
max     20152.0     100.000000        241.000000    100.000000   

       COLE_COD_MCPIO_UBICACION  ESTU_COD_RESIDE_MCPIO  ESTU_ZONA_RESIDE  \
count             545730.000000          571797.000000     573128.000000   
mean               32780.145220           33151.018797          8.470052   
std                26577.340528           26720.947192          2.633652   
min                 5001.000000            5001.000000          1.000000   
25%                11001.000000           11001.000000          8.000000   
50%                20175.000000           20383.000000         10.000000   
75%                54001.000000           54405.000000         10.000000   
max                99773.000000           99999.000000         10.000000   

       COLE_COD_DANE_INSTITUCION  COLE_COD_ICFES  ESTU_COD_RESIDE_DEPTO  \
count               5.457300e+05   545730.000000          571797.000000   
mean                2.058721e+11    80431.808964              34.319070   
std                 9.614757e+10    54570.625530             374.867568   
min                 1.050010e+11       18.000000               5.000000   
25%                 1.170500e+11    29140.000000              11.000000   
50%                 1.730010e+11    79848.000000              20.000000   
75%                 3.052660e+11   124867.000000              54.000000   
max                 8.548740e+11   631622.000000           99999.000000   

           ...        ESTU_COD_MCPIO_PRESENTACION  PUNT_LECTURA_CRITICA  \
count      ...                      573128.000000         573128.000000   
mean       ...                       33092.287023             49.779252   
std        ...                       26733.348544              9.382948   
min        ...                        5001.000000              0.000000   
25%        ...                       11001.000000             43.000000   
50%        ...                       20175.000000             49.000000   
75%        ...                       54405.000000             56.000000   
max        ...                       99773.000000            100.000000   

       PUNT_MATEMATICAS  PUNT_C_NATURALES  PUNT_SOCIALES_CIUDADANAS  \
count     573128.000000     573128.000000             573128.000000   
mean          50.233653         50.117141                 49.889524   
std           12.121206         10.053783                 11.140041   
min            0.000000          0.000000                  0.000000   
25%           42.000000         43.000000                 43.000000   
50%           49.000000         49.000000                 50.000000   
75%           57.000000         56.000000                 57.000000   
max          100.000000        100.000000                100.000000   

         PUNT_INGLES  PUNT_RAZONA_CUANT  PUNT_COMP_CIUDADANA    PUNT_GLOBAL  \
count  573128.000000      573128.000000        573128.000000  573128.000000   
mean       50.536327          51.221321            49.410981     250.229204   
std        10.916471          13.472122             9.892985      47.132512   
min         0.000000           0.000000             0.000000       0.000000   
25%        43.000000          42.000000            43.000000     217.000000   
50%        49.000000          49.000000            50.000000     245.000000   
75%        54.000000          59.000000            57.000000     278.000000   
max       100.000000         100.000000           100.000000     492.000000   

         ESTU_PUESTO  
c

# Porcentaje de acuerdo a los estudiantes Matriculados

In [3]:
Matriculados_Prime = pd.read_csv("../DatosSPADIES/Matriculados_PrimerCurso_2016.csv", delimiter=";")

In [4]:
#descriptor para ver por encima las dimensiones del fichero
Matriculados_Prime.describe()

Codigo_Institucion     IES_PADRE  ID_Sector_IES   ID_Caracter  \
count        42067.000000  42067.000000   42067.000000  42067.000000   
mean          3024.239285   3018.346281       1.527421      3.374474   
std           2456.660472   2451.290435       0.499253      0.922390   
min           1101.000000   1101.000000       1.000000      1.000000   
25%           1706.000000   1706.000000       1.000000      3.000000   
50%           2102.000000   2102.000000       2.000000      4.000000   
75%           2848.000000   2848.000000       2.000000      4.000000   
max           9915.000000   9915.000000       2.000000      4.000000   

       Codigo_departamento(IES)  Codigo_Municipio(IES)  Codigo_SNIES_programa  \
count              42067.000000           42067.000000           42067.000000   
mean                  22.721516           22750.035443           53198.567856   
std                   23.382128           23403.877900           41902.566765   
min                    5.000000            5001.000000               1.000000   
25%                   11.000000           11001.000000            8093.000000   
50%                   11.000000           11001.000000           53276.000000   
75%                   19.000000           19001.000000          101493.000000   
max                   91.000000           91001.000000          106062.000000   

       ID_Nivel_Academico  ID_Nivel_Formacion  ID_Metodologia       ID_Area  \
count        42067.000000        42067.000000    42067.000000  42067.000000   
mean             1.276559            4.600114        1.332731      5.730382   
std              0.447302            2.047185        0.631729      1.968516   
min              1.000000            1.000000        1.000000      1.000000   
25%              1.000000            3.000000        1.000000      5.000000   
50%              1.000000            5.000000        1.000000      6.000000   
75%              2.000000            6.000000        1.000000      8.000000   
max              2.000000           10.000000        3.000000      9.000000   

       Codigo_Departamento(Programa)  Codigo_Municipio(Programa)  \
count                   42067.000000                42067.000000   
mean                       31.675993                31778.438848   
std                        27.131222                27166.130041   
min                         5.000000                 5001.000000   
25%                        11.000000                11001.000000   
50%                        17.000000                17001.000000   
75%                        63.000000                63001.000000   
max                        99.000000                99001.000000   

            ID_Sexo     Anio      Semestre  Primer_Curso_2016  
count  42067.000000  42067.0  42067.000000       42067.000000  
mean       1.494378   2016.0      1.505883          22.654052  
std        0.499974      0.0      0.499971          65.720122  
min        1.000000   2016.0      1.000000           1.000000  
25%        1.000000   2016.0      1.000000           3.000000  
50%        1.000000   2016.0      2.000000          10.000000  
75%        2.000000   2016.0      2.000000          24.000000  
max        2.000000   2016.0      2.000000        4265.000000

In [5]:
Matriculados_Prime.columns

Index(['Codigo_Institucion', 'IES_PADRE',
       'Institucion_Educacion_Superior(IES)', 'Principal_o_Seccional',
       'ID_Sector_IES', 'Sector_IES', 'ID_Caracter', 'Caracter_IES',
       'Codigo_departamento(IES)', 'Departamento_domicilio_IES',
       'Codigo_Municipio(IES)', 'Municipio_domicilio_IES',
       'Codigo_SNIES_programa', 'Programa_Academico', 'ID_Nivel_Academico',
       'Nivel_Academico', 'ID_Nivel_Formacion', 'Nivel_Formacion',
       'ID_Metodologia', 'Metodologia', 'ID_Area', 'Area_Conocimiento',
       'Id_Nucleo', 'Nucleo_Basico_Conocimiento(NBC)',
       'Codigo_Departamento(Programa)', 'Departamento_oferta_programa',
       'Codigo_Municipio(Programa)', 'Municipio_oferta_del_programa',
       'ID_Sexo', 'Sexo', 'Anio', 'Semestre', 'Primer_Curso_2016'],
      dtype='object')

In [13]:
#un head para ver ejemplos de los datos 
Matriculados_Prime.head(3)

Codigo_Institucion  IES_PADRE Institucion_Educacion_Superior(IES)  \
0                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
1                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
2                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   

  Principal_o_Seccional  ID_Sector_IES Sector_IES  ID_Caracter Caracter_IES  \
0             PRINCIPAL              1    Oficial            4  Universidad   
1             PRINCIPAL              1    Oficial            4  Universidad   
2             PRINCIPAL              1    Oficial            4  Universidad   

   Codigo_departamento(IES) Departamento_domicilio_IES        ...         \
0                        11                 Bogotá D.C        ...          
1                        11                 Bogotá D.C        ...          
2                        11                 Bogotá D.C        ...          

            Nucleo_Basico_Conocimiento(NBC) Codigo_Departamento(Programa)  \
0  Ingeniería Agronómica, Pecuaria y Afines                            11   
1                      Medicina Veterinaria                            11   
2                                 Zootecnia                            11   

   Departamento_oferta_programa Codigo_Municipio(Programa)  \
0                    Bogotá D.C                      11001   
1                    Bogotá D.C                      11001   
2                    Bogotá D.C                      11001   

   Municipio_oferta_del_programa ID_Sexo    Sexo  Anio  Semestre  \
0                         Bogota       1  HOMBRE  2016         1   
1                         Bogota       1  HOMBRE  2016         1   
2                         Bogota       1  HOMBRE  2016         1   

  Primer_Curso_2016  
0                46  
1                21  
2                15  

[3 rows x 33 columns]

##  Total de Matriculados en el 2016 de todas las sedes de todas las universidades

In [15]:
sum(Matriculados_Prime["Primer_Curso_2016"])

952988

Se puede agrupar los datos por institucion asi:

In [7]:
Matriculados_groups = Matriculados_Prime.groupby('Codigo_Institucion')

In [8]:
Matriculados_groups.groups.keys()

dict_keys([2847, 2830, 1707, 4101, 4102, 1722, 3103, 1708, 4106, 4107, 4108, 4109, 4110, 2733, 4112, 9902, 9906, 9903, 3102, 2848, 2831, 2736, 3107, 1723, 4726, 2737, 3114, 3115, 3116, 2738, 4701, 4709, 1715, 4702, 2102, 2825, 2104, 2740, 2106, 2849, 2110, 2741, 2114, 2827, 1718, 4111, 2828, 1207, 3719, 1101, 1102, 1103, 1104, 2744, 1106, 1107, 1108, 1109, 1110, 1111, 1112, 1113, 1114, 1115, 1117, 1118, 1119, 1120, 1121, 1122, 1123, 4708, 1125, 1126, 2833, 1212, 4714, 2834, 2749, 4721, 1214, 3702, 3703, 2836, 3706, 1215, 3710, 1813, 3712, 1728, 3715, 3204, 2838, 3718, 1729, 3720, 5801, 3724, 1730, 2702, 9910, 2704, 2840, 2707, 2708, 2710, 2711, 2712, 2713, 2715, 2842, 2206, 1733, 2720, 2209, 2211, 2724, 1701, 1702, 1703, 1704, 1705, 2730, 2731, 2732, 1709, 1710, 1711, 1712, 1713, 1202, 1203, 1204, 1717, 1206, 1719, 1208, 2745, 2746, 2747, 1724, 1725, 1726, 1727, 1216, 1217, 1218, 1219, 1220, 1221, 1734, 1223, 4808, 4810, 4727, 4812, 2850, 4813, 4817, 4818, 2837, 4822, 3705, 3801, 4826,

In [9]:
print("Hay:", len(Matriculados_groups.groups.keys()), "grupos (sedes de Educacion Superior)")

Hay: 316 grupos (sedes de Educacion Superior)


Se puede utilizar la siguiente funcion para obtener los grupos segun la institucion y ademas de eso los parametros que nos interezan y la cantidad de matriculados total

In [10]:
def get_enrolled(df):
    return [set(df["Institucion_Educacion_Superior(IES)"]), set("Principal_o_Seccional"), set(df["ID_Sector_IES"]), set(df["Sector_IES"]), set(df["ID_Caracter"]), set(df["Caracter_IES"]), set(df["Codigo_departamento(IES)"]), set(df["Departamento_domicilio_IES"]), set(df["Codigo_Municipio(IES)"]), set(df["Municipio_domicilio_IES"]), set(df["ID_Sexo"]), set(df["Sexo"]), set(df["Semestre"]), np.sum(df["Primer_Curso_2016"])]

Se puede agrupar por el nombre de la universidad, de esta manera no importa si es la sede principal o no, se obtiene un grupo por cada universidad incluyendo todas las sedes.  
De esta manera podemos reducir el numero de grupos de 320 a 274

In [11]:
Matriculados_groups_info = Matriculados_Prime.groupby('Institucion_Educacion_Superior(IES)')

In [12]:
len(Matriculados_groups_info.groups.keys())

272

In [13]:
tmp = Matriculados_groups_info.apply(get_enrolled)

In [14]:
tmp[0]

[{'CENTRO DE EDUCACION MILITAR - CEMIL'},
 {'P', 'S', '_', 'a', 'c', 'e', 'i', 'l', 'n', 'o', 'p', 'r'},
 {1},
 {'Oficial'},
 {3},
 {'Institución Universitaria/Escuela Tecnológica'},
 {11},
 {'Bogotá D.C'},
 {11001},
 {'Bogota'},
 {1, 2},
 {'HOMBRE', 'MUJER'},
 {1, 2},
 1135]

Busco una manera de ver todas los posibles valores de las columnas asi:

In [15]:
set(Matriculados_Prime["Nivel_Formacion"])

{'Doctorado',
 'Especialización Médico Quirúrgica',
 'Especialización Tecnológica',
 'Especialización Técnico Profesional',
 'Especialización Universitaria',
 'Formación Técnica Profesional',
 'Maestría',
 'Tecnológica',
 'Universitaria'}

Existen varios niveles de formacion, y los que me interesan son los de nivel Universitario.  

Entonces se decide agrupar por nivel de formacion:

In [16]:
Matriculados_groups_info = Matriculados_Prime.groupby('Nivel_Formacion')

In [17]:
Matriculados_groups_info.groups.keys()

dict_keys(['Maestría', 'Especialización Técnico Profesional', 'Especialización Tecnológica', 'Universitaria', 'Especialización Médico Quirúrgica', 'Especialización Universitaria', 'Tecnológica', 'Formación Técnica Profesional', 'Doctorado'])

In [18]:
#Guardo en una variable el grupo
University_NewStudents = Matriculados_groups_info.get_group('Universitaria')

Obtengo el total de estudiantes que entraron a la universidad asi:

In [19]:
sum(University_NewStudents["Primer_Curso_2016"])

482917

Para conocer el porcentaje que se busca, necesito dividir esta cifra en los dos semestres del año.

In [20]:
Segun_Semestre = University_NewStudents.groupby('Semestre')

In [21]:
Segun_Semestre.groups.keys()

dict_keys([1, 2])

Guardo el grupo del primer semestre en una variable para poder tratar los datos

In [22]:
First_Semester = Segun_Semestre.get_group(1)

Obtengo la cantidad (suma) de los estudiantes matriculados en primer curso, en el primer semestre del 2016

In [23]:
print("Se matricularon:", sum(First_Semester["Primer_Curso_2016"]), "personas a primer curso en el semestre 1 del 2016")

Se matricularon: 272933 personas a primer curso en el semestre 1 del 2016


Ahora calculare la cantidad de personas que hicieron el ICFES en el año 2015 en el semestre 2

In [54]:
ICFES_2015.shape

(573128, 86)

### <span style="color: red;"> Entonces podemos decir que el porcentaje de estudiantes que realizan el ICFES, y se matriculan en una institucion de educacion superior es:</span>

In [26]:
Percent = (sum(First_Semester["Primer_Curso_2016"]) * 100)/ICFES_2015.shape[0]

In [44]:
print("El " + str(Percent)[:5] + "% de los estudiantes ingresa a la educacion superior.")

El 47.62% de los estudiantes ingresa a la educacion superior.


______

______

______

# Porcentaje de acuerdo a los estudiantes Inscritos

In [14]:
Inscribed_prime = pd.read_csv("../DatosSPADIES/Inscritos_2016_15May2017.csv", delimiter = ";")

Hay 6 columnas vacias al final.  
Se eliminaran

In [15]:
Inscribed_prime.pop("Unnamed: 33");
Inscribed_prime.pop("Unnamed: 34");
Inscribed_prime.pop("Unnamed: 35");
Inscribed_prime.pop("Unnamed: 36");
Inscribed_prime.pop("Unnamed: 37");
Inscribed_prime.pop("Unnamed: 38");

In [16]:
Inscribed_prime.describe()

Codigo_Institucion     IES_PADRE  ID_Sector_IES   ID_Caracter  \
count        42764.000000  42764.000000   42764.000000  42764.000000   
mean          3035.087620   3027.675638       1.512511      3.439225   
std           2487.893491   2481.121511       0.499849      0.829403   
min           1101.000000   1101.000000       1.000000      1.000000   
25%           1710.000000   1710.000000       1.000000      3.000000   
50%           2102.000000   2102.000000       2.000000      4.000000   
75%           2833.000000   2833.000000       2.000000      4.000000   
max           9920.000000   9915.000000       2.000000      4.000000   

       Codigo_departamento(IES)  Codigo_Municipio(IES)       ID_Sexo     Anio  \
count              42764.000000           42764.000000  42764.000000  42764.0   
mean                  21.608339           21638.865050      1.492236   2016.0   
std                   22.586452           22610.135506      0.499946      0.0   
min                    5.000000            5001.000000      1.000000   2016.0   
25%                   11.000000           11001.000000      1.000000   2016.0   
50%                   11.000000           11001.000000      1.000000   2016.0   
75%                   17.000000           17001.000000      2.000000   2016.0   
max                   91.000000           91001.000000      2.000000   2016.0   

           Semestre  Inscritos_2016  
count  42764.000000    42764.000000  
mean       1.476896       50.369563  
std        0.499472      199.801859  
min        1.000000        1.000000  
25%        1.000000        4.000000  
50%        1.000000       14.000000  
75%        2.000000       40.000000  
max        2.000000    14961.000000

In [17]:
Inscribed_prime.columns

Index(['Codigo_Institucion', 'IES_PADRE',
       'Institucion_Educacion_Superior(IES)', 'Principal_o_Seccional',
       'ID_Sector_IES', 'Sector_IES', 'ID_Caracter', 'Caracter_IES',
       'Codigo_departamento(IES)', 'Departamento_domicilio_IES',
       'Codigo_Municipio(IES)', 'Municipio_domicilio_IES',
       'Codigo_SNIES_programa', 'Programa_Academico', 'ID_Nivel_Academico',
       'Nivel_Academico', 'ID_Nivel_Formacion', 'Nivel_Formacion',
       'ID_Metodologia', 'Metodologia', 'ID_Area', 'Area_Conocimiento',
       'Id_Nucleo', 'Nucleo_Basico_Conocimiento(NBC)',
       'Codigo_Departamento(Programa)', 'Departamento_oferta_programa',
       'Codigo_Municipio(Programa)', 'Municipio_oferta_programa', 'ID_Sexo',
       'Sexo', 'Anio', 'Semestre', 'Inscritos_2016'],
      dtype='object')

In [27]:
set(Inscribed_prime["Nivel_Formacion"])

{'Doctorado',
 'Especialización Médico Quirúrgica',
 'Especialización Tecnológica',
 'Especialización Técnico Profesional',
 'Especialización Universitaria',
 'Formación Técnica Profesional',
 'Maestría',
 'No aplica',
 'Tecnológica',
 'Universitaria'}

In [28]:
#hago grupos por el nivel de formacion
Inscribed_groups = Inscribed_prime.groupby("Nivel_Formacion")

In [34]:
#Tomo el grupo de nivel de formacion Universitaria
Inscribed_U = Inscribed_groups.get_group("Universitaria")

In [35]:
Inscribed_U.describe()

Codigo_Institucion     IES_PADRE  ID_Sector_IES   ID_Caracter  \
count        17634.000000  17634.000000   17634.000000  17634.000000   
mean          2174.216570   2162.618011       1.527447      3.721334   
std           1276.281833   1247.154107       0.499260      0.504538   
min           1101.000000   1101.000000       1.000000      1.000000   
25%           1704.000000   1704.000000       1.000000      3.000000   
50%           1832.000000   1832.000000       2.000000      4.000000   
75%           2719.000000   2719.000000       2.000000      4.000000   
max           9915.000000   9915.000000       2.000000      4.000000   

       Codigo_departamento(IES)  Codigo_Municipio(IES)       ID_Sexo     Anio  \
count              17634.000000           17634.000000  17634.000000  17634.0   
mean                  24.290121           24327.298458      1.498185   2016.0   
std                   23.967160           23995.609180      0.500011      0.0   
min                    5.000000            5001.000000      1.000000   2016.0   
25%                   11.000000           11001.000000      1.000000   2016.0   
50%                   11.000000           11001.000000      1.000000   2016.0   
75%                   27.000000           27001.000000      2.000000   2016.0   
max                   86.000000           86001.000000      2.000000   2016.0   

           Semestre  Inscritos_2016  
count  17634.000000    17634.000000  
mean       1.471929       57.768685  
std        0.499226      146.347559  
min        1.000000        1.000000  
25%        1.000000        6.000000  
50%        1.000000       20.000000  
75%        2.000000       57.000000  
max        2.000000     6134.000000

In [37]:
#Hago Grupos por semestre del año
First_Semester_Inscribed = Inscribed_U.groupby("Semestre")
First_Semester_Inscribed.groups.keys()

dict_keys([1, 2])

Teniendo el grupo del semestre 1 del año 2016, de los inscritos a un nivel universitario, HALLO la suma de los inscritos.

In [43]:
Inscribed_2016 = sum(First_Semester_Inscribed.get_group(1)["Inscritos_2016"])

In [83]:
print("Se inscribieron:", Inscribed_2016, "al primer semestre del año 2016 a la Universidad")

Se inscribieron: 618512 al primer semestre del año 2016 a la Universidad


### <span style="color: red">Entonces hallo el porcentaje deseado asi:</span>

In [84]:
Percent1 = (Inscribed_2016 * 100)/ICFES_2015.shape[0]

In [85]:
print("El " + str(Percent1)[:5] + "% de los estudiantes ingresa a la educacion superior.")

El 107.9% de los estudiantes ingresa a la educacion superior.


# Porcentaje de acuerdo a los estudiantes Admitidos

In [69]:
Admitted_prime = pd.read_csv("../DatosSPADIES/admitidos2016.csv", delimiter = ";")

In [70]:
Admitted_prime.head(3)

Codigo_Institucion  IES_PADRE Institucion_Educacion_Superior(IES)  \
0                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
1                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
2                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   

  Principal_o_Seccional  ID_Sector_IES Sector_IES  ID_Caracter_IES  \
0             PRINCIPAL              1    OFICIAL                4   
1             PRINCIPAL              1    OFICIAL                4   
2             PRINCIPAL              1    OFICIAL                4   

  Caracter_IES  Codigo_departamento(IES) Departamento_domicilio_IES  \
0  Universidad                        11                 BOGOTA D.C   
1  Universidad                        11                 BOGOTA D.C   
2  Universidad                        11                 BOGOTA D.C   

      ...        Sexo  Anio  Semestre Admitidos_2016  Unnamed: 33 Unnamed: 34  \
0     ...      HOMBRE  2016         1             46          NaN         NaN   
1     ...      HOMBRE  2016         2             91          NaN         NaN   
2     ...       MUJER  2016         1             27          NaN         NaN   

   Unnamed: 35 Unnamed: 36  Unnamed: 37 Unnamed: 38  
0          NaN         NaN          NaN         NaN  
1          NaN         NaN          NaN         NaN  
2          NaN         NaN          NaN         NaN  

[3 rows x 39 columns]

De nuevo se encuentran 6 columnas vacias.  
Se eliminaran enseguida

In [71]:
Admitted_prime.pop("Unnamed: 33");
Admitted_prime.pop("Unnamed: 34");
Admitted_prime.pop("Unnamed: 35");
Admitted_prime.pop("Unnamed: 36");
Admitted_prime.pop("Unnamed: 37");
Admitted_prime.pop("Unnamed: 38");

In [72]:
Admitted_prime.describe()

Codigo_Institucion     IES_PADRE  ID_Sector_IES  ID_Caracter_IES  \
count        41219.000000  41219.000000   41219.000000     41219.000000   
mean          3063.182367   3056.280696       1.528227         3.428225   
std           2531.206298   2525.135603       0.499209         0.834392   
min           1101.000000   1101.000000       1.000000         1.000000   
25%           1709.000000   1709.000000       1.000000         3.000000   
50%           2102.000000   2102.000000       2.000000         4.000000   
75%           2834.000000   2833.000000       2.000000         4.000000   
max           9915.000000   9915.000000       2.000000         4.000000   

       Codigo_departamento(IES)  Codigo_Municipio(IES)  \
count              41219.000000           41219.000000   
mean                  21.907591           21938.117276   
std                   22.969797           22992.998714   
min                    5.000000            5001.000000   
25%                   11.000000           11001.000000   
50%                   11.000000           11001.000000   
75%                   17.000000           17001.000000   
max                   91.000000           91001.000000   

       Codigo_SNIES_del_programa  ID_Nivel_Academico  ID_Nivel_Formacion  \
count               41219.000000        41219.000000        41219.000000   
mean                54343.204784            1.292438            4.520779   
std                 42574.986460            0.454888            2.084110   
min                     1.000000            1.000000            1.000000   
25%                  8322.000000            1.000000            2.000000   
50%                 53703.000000            1.000000            5.000000   
75%                101778.000000            2.000000            6.000000   
max                106062.000000            2.000000           10.000000   

       ID_Metodologia       ID_Area  Codigo_Departamento(Programa)  \
count    41219.000000  41219.000000                   41219.000000   
mean         1.319028      5.727990                      30.604018   
std          0.630988      2.002656                      26.951368   
min          1.000000      1.000000                       5.000000   
25%          1.000000      5.000000                      11.000000   
50%          1.000000      6.000000                      15.000000   
75%          1.000000      8.000000                      54.000000   
max          3.000000      9.000000                      99.000000   

       Codigo_Municipio(Programa)     Anio      Semestre  Admitidos_2016  
count                41219.000000  41219.0  41219.000000    41219.000000  
mean                 30696.959024   2016.0      1.483127       28.625537  
std                  26979.417887      0.0      0.499721       80.422039  
min                   5001.000000   2016.0      1.000000        1.000000  
25%                  11001.000000   2016.0      1.000000        4.000000  
50%                  15176.000000   2016.0      1.000000       12.000000  
75%                  54518.000000   2016.0      2.000000       30.000000  
max                  99001.000000   2016.0      2.000000     4211.000000

In [75]:
Admitted_prime.columns

Index(['Codigo_Institucion', 'IES_PADRE',
       'Institucion_Educacion_Superior(IES)', 'Principal_o_Seccional',
       'ID_Sector_IES', 'Sector_IES', 'ID_Caracter_IES', 'Caracter_IES',
       'Codigo_departamento(IES)', 'Departamento_domicilio_IES',
       'Codigo_Municipio(IES)', 'Municipio_domicilio_IES',
       'Codigo_SNIES_del_programa', 'Programa_Academico', 'ID_Nivel_Academico',
       'Nivel_Academico', 'ID_Nivel_Formacion', 'Nivel_Formacion',
       'ID_Metodologia', 'Metodologia', 'ID_Area', 'Area_Conocimiento',
       'Id_Nucleo', 'Nucleo_Basico_Conocimiento(NBC)',
       'Codigo_Departamento(Programa)', 'Departamento_de_oferta_programa',
       'Codigo_Municipio(Programa)', 'Municipio_oferta_del_programa',
       'ID_Sexo', 'Sexo', 'Anio', 'Semestre', 'Admitidos_2016'],
      dtype='object')

In [74]:
set(Admitted_prime["Nivel_Formacion"])

{'Doctorado',
 'Especialización Médico Quirúrgica',
 'Especialización Tecnológica',
 'Especialización Técnico Profesion',
 'Especialización Universitaria',
 'Formación técnica profesional',
 'Maestría',
 'Tecnológica',
 'Universitaria'}

In [76]:
Admitted_groups = Admitted_prime.groupby("Nivel_Formacion")

In [77]:
Admitted_groups.groups.keys()

dict_keys(['Formación técnica profesional', 'Especialización Universitaria', 'Tecnológica', 'Especialización Tecnológica', 'Especialización Médico Quirúrgica', 'Maestría', 'Universitaria', 'Especialización Técnico Profesion', 'Doctorado'])

In [78]:
Admitted_Nivel = Admitted_groups.get_group("Universitaria")

In [79]:
First_Admitted = Admitted_Nivel.groupby("Semestre")
First_Admitted.groups.keys()

dict_keys([1, 2])

In [81]:
Admitted_2016 = sum(First_Admitted.get_group(1)["Admitidos_2016"])

In [82]:
print("Fueron admitidos:", Admitted_2016, "al primer semestre del año 2016 a la Universidad")

Fueron admitidos: 340406 al primer semestre del año 2016 a la Universidad


### <span style="color: red" >Se halla el porcentaje:</span>

In [86]:
Percent2 = (Admitted_2016 * 100)/ICFES_2015.shape[0]

In [87]:
print("El " + str(Percent2)[:5] + "% de los estudiantes ingresa a la educacion superior.")

El 59.39% de los estudiantes ingresa a la educacion superior.


# Conclusiones  

- Los archivos tenian algunas cosas por corregir en los nombres de las columnas, todo se arreglo usando la Herramienta **VIM** de *Linux*  
- Segun el primer analisis realizado, se encontro que los estudiantes matriculados equivalen al **47.62%** de los estudiantes que realizaron el ICFES en el año 2015 en el segundo semestre.
- En el analisis realizado a los Inscritos se encuentra que la cantidad de personas que se inscriben es mayor a los que presentaron el ICFES en el semestre anterior, y se encuentra un porcentaje ilogico (**107.9%**) la razon de esto es que hay personas que presentaron el ICFES en otros semestres y se inscribieron a la universidad en este por diversas razones (tales como que no fueron aceptados en semestres anteriores y lo quieren intentar de nuevo), por tanto se descarta este dato encontrado  
- Segun lo encontrado en el tercer analisis los admitidos en la universidad en el año 2016 equivalen al **59.39%** de los que realizaron el ICFES en el semestre 2 del año 2015